crates/re_log_types/src/serde_field.rs

```
@@ -0,0 +1,121 @@
+use arrow2::{
+    array::{BinaryArray, MutableBinaryArray, TryPush},
+    datatypes::DataType,
+};
+use arrow2_convert::{deserialize::ArrowDeserialize, field::ArrowField, serialize::ArrowSerialize};
+
+/// Helper for storing arbitrary serde-compatible types in an arrow2 [`BinaryArray`] field
+///
+/// Use as:
+/// ```
+/// use re_log_types::serde_field::SerdeField;
+/// use arrow2_convert::{ArrowDeserialize, ArrowField, ArrowSerialize, field::ArrowField};
+/// use arrow2::datatypes::{DataType, Field};
+///
+/// #[derive(Clone, Debug, PartialEq, serde::Deserialize, serde::Serialize)]
+/// struct SomeStruct {
+///     foo: String,
+///     bar: u32,
+/// }
+///
+/// #[derive(Clone, Debug, PartialEq, ArrowField, ArrowSerialize, ArrowDeserialize)]
+/// #[arrow_field(transparent)]
+/// struct SomeStructArrow(#[arrow_field(type = "SerdeField<SomeStruct>")] SomeStruct);
+///
+/// assert_eq!(SomeStructArrow::data_type(), DataType::Binary);
+/// ```
+pub struct SerdeField<T>(std::marker::PhantomData<T>);
+
+impl<T> ArrowField for SerdeField<T>
+where
+    T: serde::ser::Serialize + serde::de::DeserializeOwned,
+{
+    type Type = T;
+
+    #[inline]
+    fn data_type() -> DataType {
+        arrow2::datatypes::DataType::Binary
+    }
+}
+
+impl<T> ArrowSerialize for SerdeField<T>
+where
+    T: serde::ser::Serialize + serde::de::DeserializeOwned,
+{
+    type MutableArrayType = MutableBinaryArray<i32>;
+    #[inline]
+    fn new_array() -> Self::MutableArrayType {
+        MutableBinaryArray::new()
+    }
+
+    fn arrow_serialize(
+        v: &<Self as ArrowField>::Type,
+        array: &mut Self::MutableArrayType,
+    ) -> arrow2::error::Result<()> {
+        crate::profile_function!();
+        let mut buf = Vec::new();
+        rmp_serde::encode::write_named(&mut buf, v).map_err(|_err| {
+            // TODO(jleibs): Could not get re_error::format() to work here
```

use `to_string()` then. ignoring the error message completely is a bad idea